plugins/ptp_operator/config/config.go

```diff
@@ -0,0 +1,216 @@
+	holdoverTimeout    int64 = 5
+)
+
+// PtpProfile ...
+type PtpProfile struct {
+	Name              *string           `json:"name"`
+	Interface         *string           `json:"interface"`
+	PtpClockThreshold PtpClockThreshold `json:"ptpClockThreshold,omitempty"`
+}
+
+// PtpClockThreshold ...
+type PtpClockThreshold struct {
+	// clock state to stay in holdover state in secs
+	HoldOverTimeout int64 `json:"holdOverTimeout,omitempty"`
+	// max offset in nano secs
+	MaxOffsetThreshold int64 `json:"maxOffsetThreshold,omitempty"`
+	// min offset in nano secs
+	MinOffsetThreshold int64 `json:"minOffsetThreshold,omitempty"`
+}
+
+// LinuxPTPConfUpdate for ptp-conf update
+type LinuxPTPConfUpdate struct {
+	lock                   sync.RWMutex
+	UpdateCh               chan bool
+	NodeProfiles           []PtpProfile
+	appliedNodeProfileJSON []byte
+	profilePath            string
+	intervalUpdate         int
+	EventThreshold         map[string]*PtpClockThreshold
+}
+
+// NewLinuxPTPConfUpdate -- profile updater
+func NewLinuxPTPConfUpdate() *LinuxPTPConfUpdate {
+	ptpProfileUpdate := &LinuxPTPConfUpdate{
+		lock:           sync.RWMutex{},
+		UpdateCh:       make(chan bool),
+		profilePath:    DefaultProfilePath,
+		intervalUpdate: DefaultUpdateInterval,
+		EventThreshold: make(map[string]*PtpClockThreshold),
+	}
+
+	if os.Getenv("PTP_PROFILE_PATH") != "" {
+		ptpProfileUpdate.profilePath = os.Getenv("PTP_PROFILE_PATH")
+	}
+
+	if os.Getenv("CONFIG_UPDATE_INTERVAL") != "" {
+		if i := common.GetIntEnv("CONFIG_UPDATE_INTERVAL"); i > 0 {
+			ptpProfileUpdate.intervalUpdate = i
+		}
+	}
+	log.Infof("profile path set to  %s", ptpProfileUpdate.profilePath)
+	log.Infof("config update interval is set to %d secs ", ptpProfileUpdate.intervalUpdate)
+	return ptpProfileUpdate
+}
+
+// SetDefaultPTPThreshold ... creates default record
+func (l *LinuxPTPConfUpdate) SetDefaultPTPThreshold(iface string) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	l.EventThreshold[iface] = &PtpClockThreshold{
+		HoldOverTimeout:    holdoverTimeout,
+		MaxOffsetThreshold: maxOffsetThreshold,
+		MinOffsetThreshold: minOffsetThreshold,
+	}
+}
+
+// DeletePTPThreshold ... delete threshold for the interface
+func (l *LinuxPTPConfUpdate) DeletePTPThreshold(iface string) {
+	if _, found := l.EventThreshold[iface]; found {
+		l.lock.Lock()
+		defer l.lock.Unlock()
+		delete(l.EventThreshold, iface)
+	}
+}
+
+// UpdatePTPThreshold .. update ptp threshold
+func (l *LinuxPTPConfUpdate) UpdatePTPThreshold() {
+	for _, profile := range l.NodeProfiles {
+		if _, found := l.EventThreshold[*profile.Interface]; !found {
+			l.SetDefaultPTPThreshold(*profile.Interface)
+		}
+		if t, found := l.EventThreshold[*profile.Interface]; found {
+			if profile.PtpClockThreshold.MaxOffsetThreshold != 0 {
+				t.MaxOffsetThreshold = profile.PtpClockThreshold.MaxOffsetThreshold
+			}
+			if profile.PtpClockThreshold.MinOffsetThreshold != 0 {
+				t.MinOffsetThreshold = profile.PtpClockThreshold.MinOffsetThreshold
```

Comment:
Is it possible we want to update MinOffsetThreshold or HoldOverTimeout to `0`?

Reply:
not with holdover time. it is secs and min is 1secs to the max can be anything.
the minOffsetThreshold can be zero, but zero is a perfect offset which is hard to achieve.
I could remove the check of min and max threshold

Reply:
Changed the logic a bit to check for default values